### DISCUSSION TITLE
Can DatabaseQueue::getLockForPopping() use anything other than PDO::ATTR_DRIVER_NAME.

### DISCUSSION THREAD

**Discussion by leenooks** [laravel/framework#40431] [Ideas]
- Laravel Version: 8.79.0
- PHP Version: 8.0.13
- Database Driver & Version: CockroachDB (pdo_pgsql)

### Description:
I'm using CockroachDB with Laravel - which uses the PostgreSQL PDO driver. However, according to https://github.com/cockroachdb/cockroach/issues/40476 it does not support `SKIP LOCKED` (yet anyway).

I've been enhancing nbj/cockroachdb-laravel (my fork is  [leenooks/cockroachdb-laravel](https://github.com/leenooks/cockroachdb-laravel)), and what I cannot figure out is an easy way to ensure that `DatabaseQueue::getLockForPopping()` passess through the DB checks and returns true. Unfortunately that routine is using `PDO::ATTR_DRIVER_NAME` to determine what the DB is, but in this case it's not actually PostgreSQL.

Ideally, this routine should use something else PDO/Driver/... to determine the driver for the database and make its decision to include `SKIP LOCKED` based on the PHP driver, not the PDO driver.

Without this being changed, it doesnt seem that I can run job queues with CockroachDB.

**Comment by leenooks** [comment]
@driesvints I actually do think this is a bug.

The purpose of that function is to determine whether the database is a type that allows `SKIP LOCKED`. And it makes this determination by the PDO::ATTR_DRIVER_NAME.

The problem with this approach is that many database are using the pgsql driver to operate - but it doesn't mean that it is a PostgreSQL database, and further doesn't mean it supports `SKIP LOCKED`.

My use case example is CockroachDB, which uses the pgsql driver, but doesn't support `SKIP LOCKED`.

**Reply by driesvints** [reply]
We don't support CockroachDB, sorry. If you can't reproduce this with a vanilla pgsql DB then we can't help you.

**Comment by leenooks** [comment]
I know Laravel doesnt support CockroachDB - but using it with Laravel is pretty easy and mostly without incident.

The issue is when PDO::ATTR_DRIVER_NAME = 'pgsql' that does not mean PostgreSQL databases - which is my point. Many databases use the "postgres wire protocol", eg: CockroachDB, CrateDB, RavonDB and I know there are more than I've discovered.

If this function was changed to better determine if the database is *actually* PostgreSQL database, and thus provide the `SKIP LOCKED` capability that would be so helpful to those who want to use an alternative database (that uses the postgres wire protocol) with Laravel
